airflow/www/templates/airflow/dag.html

```diff
@@ -573,5 +581,59 @@ <h4 class="modal-title" id="dagModalLabel">
       $.post(url);
     });
 
+    // AutoRefresh
+    AUTO_REFRESH_COOKIE_NAME = 'airflow_auto_refresh_enabled'
+
+    setCookie = (cname, cvalue, exdays) => {
+    var d = new Date();
+      d.setTime(d.getTime() + (exdays*24*60*60*1000));
+      var expires = "expires="+ d.toUTCString();
+      document.cookie = cname + "=" + cvalue + ";" + expires + ";path=/";
+    }
+
+    getCookie = (cname)=> {
+      var name = cname + "=";
+      var decodedCookie = decodeURIComponent(document.cookie);
+      var ca = decodedCookie.split(';');
+      for(var i = 0; i <ca.length; i++) {
+        var c = ca[i];
+        while (c.charAt(0) == ' ') {
+          c = c.substring(1);
+        }
+        if (c.indexOf(name) == 0) {
+          return c.substring(name.length, c.length);
+        }
+      }
+      return "";
+    }
+
+    rmCookie = (cname)=>{
+      setCookie(cname, '', 0)
+    }
+
+    isAutoRefreshEnabled = ()=>{
+        return getCookie(AUTO_REFRESH_COOKIE_NAME)==1
+    }
+
+    toggleAutoRefresh = ()=>{
+      if(isAutoRefreshEnabled()){
+        $('#label-auto-refresh').html('Auto Refresh')
+        rmCookie(AUTO_REFRESH_COOKIE_NAME)
+      }else{
+        setCookie(AUTO_REFRESH_COOKIE_NAME, 1)
+      }
+    }
+
+    setInterval(()=>{
+        if (isAutoRefreshEnabled())
+            location.reload()
```

Comment:
I think what you actually want to reload is only the view that renders the nodes (d3 objects) instead of the whole page, right?

Reply:
@feluelle Yes, it could be done too. I see the refresh button uses this approach (reload the page), then I followed this idea. Actually, looking to the code I couldn't see how (or if it's possible) to only reload the view that renders the nodes. If you know a way to do that, we could try.

Reply:
In the graph view there is a button ![image](https://user-images.githubusercontent.com/23282078/64324886-74692480-cfc7-11e9-8761-ea3c3e0e1719.png) in the top right corner.

It seems to only refresh the graph view (haven't looked more into it *) which I think would be what you want.

*_you could search in the **www** or **www_rbac** (depends on whether you want to support 1.10.x or not) for `refresh_button` (the class attribute of the button)_

Reply:
As I see on the code, the endpoint returns the data in a specif format to that screen. To implement a refresh like this to the Tree View screen it is necessary to add a new endpoint (maybe use an existent, I'm not sure if we have with this format).
My suggestion is to move forward refreshing the page as a V1 implementation and add such functionality in the next version of this functionality. Also, my colleagues identified using it that we can have a time delay option to the Auto Refresh.
What is your opinion about this functionality and implementation?

Reply:
I am -1 on this. I think a page refresh should not be part of this UI.

There are browser add-ons that can do exactly this.
For example for Firefox [tab-auto-refresh](https://addons.mozilla.org/en-US/firefox/addon/tab-auto-refresh/) which _Automatically refresh tabs based on custom time intervals_.

Reply:
But I am +1 for adding auto refresh for specific html containers like the tree view or graph view.

Reply:
-1 for page refresh. There is already an endpoint exists, which is used by that that left corner refresh button (/object/task_instances).